crates/symbolicator-service/src/download/filesystem.rs

```diff
@@ -23,7 +23,7 @@ impl FilesystemDownloader {
     pub async fn download_source(
         &self,
         file_source: &FilesystemRemoteFile,
-        destination: &mut File,
+        mut destination: impl AsyncWrite + Unpin,
```

Comment:
This does not need to be `Unpin` you just have to pin the destination to the stack before copying.

```rust
        let mut destination = std::pin::pin!(destination);
        tokio::io::copy(&mut file, &mut destination).await?;
```

```suggestion
        mut destination: impl AsyncWrite,
```

Reply:
Thanks for the tip, `pin` isn't something I've used extensively so I took the easy way out.